fighters/common/src/general_statuses/turn.rs

```diff
@@ -84,60 +76,57 @@ unsafe extern "C" fn status_turn_main(fighter: &mut L2CFighterCommon) -> L2CValu
     }
     else { false };
     if !should_end {
+        let dash_stick_x: f32 = WorkModule::get_param_float(fighter.module_accessor, hash40("common"), hash40("dash_stick_x"));
+        let stick_x = fighter.global_table[STICK_X].get_f32();
+        let turn_work_lr: f32 = WorkModule::get_float(fighter.module_accessor, *FIGHTER_STATUS_TURN_WORK_FLOAT_LR);
+
         if ControlModule::check_button_trigger(fighter.module_accessor, *CONTROL_PAD_BUTTON_CSTICK_ON) {
             VarModule::on_flag(fighter.battle_object, vars::common::DISABLE_BACKDASH);
         }
         if fighter.global_table[STICK_X].get_f32() == 0.0 {
             VarModule::off_flag(fighter.battle_object, vars::common::DISABLE_BACKDASH);
         }
+        if fighter.global_table[SITUATION_KIND] == SITUATION_KIND_GROUND
+        && stick_x * -1.0 * turn_work_lr < dash_stick_x  // if left stick is below dash threshold
+        && VarModule::is_flag(fighter.battle_object, vars::common::IS_SMASH_TURN)  // AND you are currently in a smash turn
+        && StatusModule::prev_status_kind(fighter.module_accessor, 0) == *FIGHTER_STATUS_KIND_DASH  // AND your previous status was a dash (not turn)
+        && MotionModule::frame(fighter.module_accessor) <= 1.0 {  // AND you are on frame 0 or frame 1 of your smash turn
+            // perfect pivot
+            VarModule::off_flag(fighter.battle_object, vars::common::IS_SMASH_TURN);
+            let dash_speed: f32 = WorkModule::get_param_float(fighter.module_accessor, hash40("dash_speed"), 0);
+            let multiplier = -0.75;
+            let pivot_boost: smash::phx::Vector3f = smash::phx::Vector3f {x: dash_speed * multiplier, y: 0.0, z: 0.0};
+            KineticModule::clear_speed_all(fighter.module_accessor);
+            KineticModule::add_speed(fighter.module_accessor, &pivot_boost);
+        }
+        /***fighter.clear_lua_stack();
+        lua_args!(fighter, FIGHTER_KINETIC_ENERGY_ID_CONTROL);
+        let speed_control = app::sv_kinetic_energy::get_speed_x(fighter.lua_state_agent);
+        println!("turn speed_control: {}", speed_control);
+        println!("turn total speed: {}", KineticModule::get_sum_speed_x(fighter.module_accessor, *KINETIC_ENERGY_RESERVE_ATTRIBUTE_ALL) - KineticModule::get_sum_speed_x(fighter.module_accessor, *KINETIC_ENERGY_RESERVE_ATTRIBUTE_GROUND) - KineticModule::get_sum_speed_x(fighter.module_accessor, *KINETIC_ENERGY_RESERVE_ATTRIBUTE_EXTERN));
+        ***/
```

Comment:
Moved perfect pivot block before status_turncommon (which checks for transitions out of Turn) is checked. This allows your slide to first be applied before transitioning into any other status (frame-perfect pivot cancels would previously not apply slide because of this)